PokeAlarm/Manager.py

```diff
@@ -847,6 +847,14 @@ def process_egg(self, egg):
             return
 
         gym_id = egg['id']
+        gym_info = self.__cache.get_gym_info(gym_id)
+        gym_name = gym_info['name'].lower()
+
+        # Check if egg gym should be sponsored and is sponsored
+        if len(self.__egg_settings['contains']) > 0:
```

Comment:
This should be after processed check because we don't want to keep getting notified about the same egg getting rejected over and over - I would suggest moving it to right before geofences.